pkg/apis/samplesoperator/v1alpha1/types.go

```diff
@@ -47,30 +55,26 @@ type SamplesResourceSpec struct {
 type SamplesResourceStatus struct {
 	// Conditions represents the available maintenance status of the sample
 	// imagestreams and templates.
-	Conditions []SamplesResourceCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,8,rep,name=conditions"`
+	Conditions    []ResourceCondition
+	SecretCreated bool
 }
 
-type SamplesResourceConditionType string
+type ConditionType string
 
 // the valid conditions of the SamplesResource
 
 const (
-	SamplesUpToDate SamplesResourceConditionType = "UpToDate"
-
-	SamplesUpdateFailed SamplesResourceConditionType = "UpdateFailed"
+	SamplesUpdateComplete ConditionType = "SamplesResourceUpdateComplete"
+	SamplesUpdateFailed   ConditionType = "SamplesResourceUpdateFailed"
+	SecretUpdateComplete  ConditionType = "SecretUpdateComplete"
+	SecretUpdateFailed    ConditionType = "SecretUpdateFailed"
 )
 
-type SamplesResourceCondition struct {
+type ResourceCondition struct {
 	// Type of condition.
-	Type SamplesResourceConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=SamplesResourceConditionType"`
-	// Status of the condition, one of True, False, Unknown.
-	Status corev1.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=k8s.io/kubernetes/pkg/api/v1.ConditionStatus"`
-	// The last time this condition was updated.
-	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty" protobuf:"bytes,6,opt,name=lastUpdateTime"`
-	// The last time the condition transitioned from one status to another.
-	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,3,opt,name=lastTransitionTime"`
-	// The reason for the condition's last transition.
-	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	Type ConditionType
+	// The time this condition occurred
+	Time metav1.Time
```

Comment:
any reason for not following the pattern used elsewhere (w/ lastupdatetime, lasttransition, etc)?  I think the condition pattern was done that way to make it easier for cli tools to handle any object that exposed conditions, in a consistent way, so we probably want to follow it.

Reply:
It seemed overkill in that this is *NOT* a FSM like deployments.  In my mind these are event logs that can occur multiple times, independent of each other.

Perhaps I should call change the use of Condition to Event with the various fields/types to avoid confusion?

Reply:
logging events (via the event recorder) is not entirely unreasonably, but i'd say that's orthogonal to the fact that we need to have conditions that reflect the current state/health of the operator.

Conditions I think we should have:
ImportCredentialsExist true/false
SamplesExist true/false